### PR TITLE
unittests: enable ASAN on native

### PR DESCRIPTION
**Pull request by kaspar030**
### Contribution description

This PR wires up the Address Sanitizer for the unittests on native.

Another ~~PR~~ commit fixes the last remaining issue ASAN reported (a scratch buffer was too small in tests-uri_parser).

### Testing procedure

This only affects native, and makes it running unittests more likely to fail. CI tests all of it.

### Issues/PRs references

 #17800

### CODE REVIEW

File: tests/unittests/tests-uri_parser/tests-uri_parser.c [hunk 1]
```diff
@@ -48,8 +48,7 @@
         }                                                                   \
     } while (0)
 
-#define VEC_MSG_LEN (sizeof("Unexpected userinfo member \"\" for \"\"") + \
-                     64U + 8U)
+#define VEC_MSG_LEN   (256)
```
Comment (kaspar030) [inline tests/unittests/tests-uri_parser/tests-uri_parser.c:51]: This was like ~110 before. tests are executed pretty low on the stack, so this *should* be ok.

**Comment by kaspar030** [comment]
(unrelated build failure in `tests/gnrc_netif_ieee802154`)

File: tests/unittests/tests-uri_parser/tests-uri_parser.c (hunk 1 above)
Comment (miri64) [reply tests/unittests/tests-uri_parser/tests-uri_parser.c:51]: Can you tell which test exactly tripped the asan? The original `VEC_MSG_LEN` might be based on a wrong assumption, which in testing is always a warning sign.

Comment (miri64) [reply tests/unittests/tests-uri_parser/tests-uri_parser.c:51]: (so I'd rather fix the value for the correct assumption than just making it large enough, to account for exactly those kind of wrong assumptions)

Comment (miri64) [reply tests/unittests/tests-uri_parser/tests-uri_parser.c:51]: Ah... Just saw, that this is just for the output messages of the assert... so nvm.

Comment (kaspar030) [reply tests/unittests/tests-uri_parser/tests-uri_parser.c:51]: jup, my debugging went the same way, but it was actually the first test (so first message) where it tripped. Then I tried to parse the macro for why this fails, but that was not very fruitful.

**Review by miri64 — APPROVED** [review]
Looking good. ACK

**Pull request merged**